packages/mongodb-ts-autocomplete/src/index.ts

```diff
@@ -0,0 +1,209 @@
+import Autocompleter from '@mongodb-js/ts-autocomplete';
+import type {
+  AutocompleterOptions,
+  AutoCompletion,
+} from '@mongodb-js/ts-autocomplete';
+import autocompleteTypes from './fixtures/autocomplete-types';
+
+import type { JSONSchema } from './type-export';
+import { toTypescriptTypeDefinition } from './type-export';
+
+import {
+  inferCollectionNameFromFunctionCall,
+  compileSourceFile,
+} from './cdt-analyser';
+
+import { CachingAutocompletionContext } from './autocompletion-context';
+import type { AutocompletionContext } from './autocompletion-context';
+
+export { JSONSchema } from './type-export';
+export { AutocompletionContext } from './autocompletion-context';
+
+type MongoDBAutocompleterOptions = {
+  context: AutocompletionContext;
+  autocompleterOptions?: AutocompleterOptions;
+};
+
+class DatabaseSchema {
+  private collectionSchemas: Record<string, JSONSchema | undefined>;
+
+  constructor() {
+    // TODO: this is kinda the only real reason for this class: So we can keep
```

Comment:
I'll remove the TODO before merging, just keen on hearing people's thoughts first.